FreeRTOS/FreeRTOS
# Fix stack smashing in POSIX Demo

Description
-----------
The tracelyzer code was declaring a local stack variable of type `uint32_t` and passing it as output parameter to a function which expected `uint64_t` on a 64-bit platform. This resulted in 4 byte memory corruption. The problem was that the function signature uses `TraceUnsignedBaseType_t` type for the output parameter which gets defined to `uint32_t` on a 32-bit platform and to `uint64_t` on a 64-bit platform, while the local stack variable was declared as `uint32_t`. This commit changes the type of local stack variable to `TraceUnsignedBaseType_t` to match the function parameter type.

Tracelyzer creates a task internally stack size of which is controlled using `TRC_CFG_CTRL_TASK_STACK_SIZE`. It was earlier defined to `1024`. The POSIX port requires the stack size to be minimum `PTHREAD_STACK_MIN`. The commit updates the `TRC_CFG_CTRL_TASK_STACK_SIZE` to `PTHREAD_STACK_MIN`.

The demo directory contained one extra `trcConfig.h` and `trcKernelPortConfig.h`. This commit deletes those files.

Test Steps
-----------
Output before - 
```
[WARN] pthread_attr_setstack failed with return value: 22. Default stack will be used.
[WARN] Increase the stack size to PTHREAD_STACK_MIN.

Trace started.
The trace will be dumped to disk if a call to configASSERT() fails.
Starting echo blinky demo
*** stack smashing detected ***: terminated
Aborted (core dumped)
```

Output after - 
```
Trace started.
The trace will be dumped to disk if a call to configASSERT() fails.
Starting echo blinky demo
Message received from task
Message received from task
Message received from task
Message received from task
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
